File: Groupe2/TP2/src/chaine.py
def longueur_chaine(chaine: str) -> int:
    # Calculer la longueur d'une chaîne.
    longueur = 0
    for _ in chaine:
        longueur += 1
    return longueur

def copier_chaine(source: str) -> str:
    # Copier une chaîne dans une nouvelle chaîne.
    return source

def concatener_chaines(chaine1: str, chaine2: str) -> str:
    # Concaténer deux chaînes.
    return chaine1 + chaine2

def main() -> None:
    chaine1 = input("Entrez la première chaîne : ")
    chaine2 = input("Entrez la deuxième chaîne : ")

    # Calculer la longueur des chaînes
    print(f"Longueur de la première chaîne : {longueur_chaine(chaine1)}")
    print(f"Longueur de la deuxième chaîne : {longueur_chaine(chaine2)}")

    # Copier la première chaîne dans une nouvelle chaîne
    copie = copier_chaine(chaine1)
    print(f"Après copie, résultat : {copie}")

    # Concaténer les deux chaînes
    resultat = concatener_chaines(chaine1, chaine2)
    print(f"Après concaténation, résultat : {resultat}")

if __name__ == "__main__":
    main()

File: Groupe2/TP2/src/chaine.c
#include <stdio.h>

// Fonction pour calculer la longueur d'une chaîne
int longueur_chaine(const char *chaine) {
    int longueur = 0;
    while (chaine[longueur] != '\0') {
        longueur++;
    }
    return longueur;
}

// Fonction pour copier une chaîne dans une autre
void copier_chaine(char *destination, const char *source) {
    int i = 0;
    while (source[i] != '\0') {
        destination[i] = source[i];
        i++;
    }
    destination[i] = '\0'; 
}

// Fonction pour concaténer deux chaînes
void concatener_chaines(char *destination, const char *source) {
    int i = 0;
    while (destination[i] != '\0') {
        i++;
    }
    int j = 0;
    while (source[j] != '\0') {
        destination[i] = source[j];
        i++;
        j++;
    }
    destination[i] = '\0';
}

int main() {
    char chaine1[100], chaine2[100];
    char result[200];

    // Demande des chaînes à l'utilisateur
    printf("Entrez la première chaîne : ");
    fgets(chaine1, sizeof(chaine1), stdin);
    printf("Entrez la deuxième chaîne : ");
    fgets(chaine2, sizeof(chaine2), stdin);

    // Enleve le caractère de nouvelle ligne s'il est présent
    chaine1[strcspn(chaine1, "\n")] = '\0';
    chaine2[strcspn(chaine2, "\n")] = '\0';

    // Calculer la longueur des chaînes
    printf("Longueur de la première chaîne : %d\n", longueur_chaine(chaine1));
    printf("Longueur de la deuxième chaîne : %d\n", longueur_chaine(chaine2));

    // Copier la première chaîne dans la deuxième chaîne
    copier_chaine(result, chaine1);
    printf("Après copie, résultat : %s\n", result);

    // Concaténer les deux chaînes
    concatener_chaines(result, chaine2);
    printf("Après concaténation, résultat : %s\n", result);

    return 0;
}
